### DISCUSSION TITLE
import transactions from email notifications

### DISCUSSION THREAD

**Discussion by blathers123** [firefly-iii/firefly-iii#7767] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

I'd like to import transactions from my bank automatically without relying on 3rd party integrations or services. My bank supports email notifications for certain events, ie I buy something with my credit card. I'd like to have firefly automatically check the email account, read in any emails from my bank, and import the relevant information.

### Solution

Add email inbox as a way to import data

**Comment by JC5** [comment, marked as answer]
This is a great idea! It's out of the scope of Firefly III though. You'll have to build it yourself or find somebody who can. The [Firefly III API](https://api-docs.firefly-iii.org/) should be combined with a script that parses the emails.

**Reply by blathers123** [reply]
Do you think it's out of scope for the data-parser as well? I figured that would be a more appropriate place for this functionality

**Reply by JC5** [reply]
The "functionality" you ask for is a project the size of Firefly III in itself.  It's a complex program to build, error-prone and it needs a thousand little user-specific tweaks to get it right. You could float more than a few full-time developers on "functionality" like this.

**Reply by blathers123** [reply]
You're not wrong, parsing text is always messy, and you'd know better than I would about how complex tracking finances are on-top of that. Add in different banks, messages, and languages and it's definitely not straight forward.  
I can see a system where a parser looks for certain key words to figure out the type. "Approved/Declined" "Deposited/withdrawn" etc. I think that would cover 90% of cases without being overly complex (I'm definitely an optimistic person when it comes to complexity). Though it would really also depend on people's tolerance on having questionable data introduced to their system. 
I'll look over the API and see if I can hack a script together that fits my needs. Thanks for your time and your work on this project!
